AzureBasicLoadBalancerUpgrade/modules/NSGCreation/NSGCreation.psm1

```diff
@@ -76,7 +76,7 @@ function NSGCreation {
                 Name                                = ($inboundNatRule.Name + "-NatRule")
                 Protocol                            = $inboundNatRule.Protocol
                 SourcePortRange                     = "*"
-                DestinationPortRange                = (($inboundNatRule.FrontendPortRangeStart).ToString() + "-" + ($inboundNatRule.FrontendPortRangeEnd).ToString())
+                DestinationPortRange                = [string]::IsNullOrEmpty($inboundNatRule.FrontendPortRangeStart) ? ($inboundNatRule.BackendPort).ToString() : (($inboundNatRule.FrontendPortRangeStart).ToString() + "-" + ($inboundNatRule.FrontendPortRangeEnd).ToString())
```

Comment:
Ternary operators are not supported in PowerShell 5.1. We should probably move to an if/then outside the security rule object creation @Welasco